src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/PooledTxsRequestor.cs

```diff
@@ -30,78 +30,64 @@ public PooledTxsRequestor(ITxPool txPool, ITxPoolConfig txPoolConfig)
         public void RequestTransactions(Action<GetPooledTransactionsMessage> send, IReadOnlyList<Hash256> hashes)
         {
             ArrayPoolList<Hash256> discoveredTxHashes = AddMarkUnknownHashes(hashes);
-
-            if (discoveredTxHashes.Count != 0)
-            {
-                RequestPooledTransactions(send, discoveredTxHashes);
-            }
+            RequestPooledTransactions(send, discoveredTxHashes);
         }
 
         public void RequestTransactionsEth66(Action<V66.Messages.GetPooledTransactionsMessage> send, IReadOnlyList<Hash256> hashes)
         {
-            using ArrayPoolList<Hash256> discoveredTxHashes = AddMarkUnknownHashes(hashes);
+            ArrayPoolList<Hash256> discoveredTxHashes = AddMarkUnknownHashes(hashes);
 
-            if (discoveredTxHashes.Count != 0)
+            if (discoveredTxHashes.Count <= MaxNumberOfTxsInOneMsg)
             {
-                if (discoveredTxHashes.Count <= MaxNumberOfTxsInOneMsg)
-                {
-                    RequestPooledTransactionsEth66(send, discoveredTxHashes);
-                }
-                else
+                RequestPooledTransactionsEth66(send, discoveredTxHashes);
+            }
+            else
+            {
+                using ArrayPoolList<Hash256> _ = discoveredTxHashes;
+                ArrayPoolList<Hash256> hashesToRequest = new(MaxNumberOfTxsInOneMsg);
+                for (int i = 0; i < discoveredTxHashes.Count; i++)
                 {
-                    ArrayPoolList<Hash256> hashesToRequest = new(MaxNumberOfTxsInOneMsg);
-                    for (int i = 0; i < discoveredTxHashes.Count; i++)
-                    {
-                        if (hashesToRequest.Count % MaxNumberOfTxsInOneMsg == 0 && hashesToRequest.Count > 0)
-                        {
-                            RequestPooledTransactionsEth66(send, hashesToRequest);
-                            hashesToRequest = new(MaxNumberOfTxsInOneMsg);
-                        }
-
-                        hashesToRequest.Add(discoveredTxHashes[i]);
-                    }
-
-                    if (hashesToRequest.Count > 0)
+                    if (hashesToRequest.Count % MaxNumberOfTxsInOneMsg == 0)
                     {
                         RequestPooledTransactionsEth66(send, hashesToRequest);
+                        hashesToRequest = new(MaxNumberOfTxsInOneMsg);
                     }
+
+                    hashesToRequest.Add(discoveredTxHashes[i]);
```

Comment:
Could we do bulk copy here instead of adding 1 by 1?

Reply:
Refactored a bit to copy whole batch at once.